core/src/main/java/io/servicecomb/core/provider/consumer/InvokerUtils.java

```diff
@@ -72,16 +76,25 @@ public static Response innerSyncInvoke(Invocation invocation) {
           String.format("invoke failed, %s", invocation.getOperationMeta().getMicroserviceQualifiedName());
       LOGGER.debug(msg, e);
       return Response.createConsumerFail(e);
+    } finally {
+      invocation.triggerFinishedEvent();
     }
   }
 
   public static void reactiveInvoke(Invocation invocation, AsyncResponse asyncResp) {
     try {
+      triggerStartedEvent(invocation);
       ReactiveResponseExecutor respExecutor = new ReactiveResponseExecutor();
       invocation.setResponseExecutor(respExecutor);
 
-      invocation.next(asyncResp);
+      AsyncResponse wrappedAsyncResponse = response -> {
+        invocation.triggerFinishedEvent();
+        asyncResp.handle(response);
+      };
+
+      invocation.next(wrappedAsyncResponse);
```

Comment:
```
      invocation.next(ar -> {
        ......
        asyncResp.handle(ar);
      });
```
your code is no problem
but maybe this is simpler

Reply:
improve it later